internal/kafka/internal/config/aws_config.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/spf13/pflag"
+)
+
+type AWSConfig struct {
+	Route53                     awsRoute53Config
+	SecretManager               awsSecretManagerConfig
+	ConfigForOSDClusterCreation awsConfigForOSDClusterCreation
+}
+
+type awsSecretManagerConfig struct {
+	AccessKey               string
+	AccessKeyFilePath       string
+	SecretAccessKey         string
+	SecretAccessKeyFilePath string
+	SecretPrefix            string
+	Region                  string
+}
+
+type awsConfigForOSDClusterCreation struct {
+	AccountID               string
+	AccountIDFilePath       string
+	AccessKey               string
+	AccessKeyFilePath       string
+	SecretAccessKey         string
+	SecretAccessKeyFilePath string
+}
+
+type awsRoute53Config struct {
+	AccessKey               string
+	AccessKeyFilePath       string
+	SecretAccessKey         string
+	SecretAccessKeyFilePath string
+}
+
+func NewAWSConfig() *AWSConfig {
+	return &AWSConfig{
+		ConfigForOSDClusterCreation: awsConfigForOSDClusterCreation{
+			AccountIDFilePath:       "secrets/aws.accountid",
+			AccessKeyFilePath:       "secrets/aws.accesskey",
+			SecretAccessKeyFilePath: "secrets/aws.secretaccesskey",
+		},
+		Route53: awsRoute53Config{
+			AccessKeyFilePath:       "secrets/aws.route53accesskey",
+			SecretAccessKeyFilePath: "secrets/aws.route53secretaccesskey",
+		},
+		SecretManager: awsSecretManagerConfig{
+			AccessKeyFilePath:       "secrets/aws-secret-manager/aws_access_key_id",
+			SecretAccessKeyFilePath: "secrets/aws-secret-manager/aws_secret_access_key",
+			Region:                  "us-east-1",
+			SecretPrefix:            "kas-fleet-manager",
+		},
+	}
+}
+
+func (c *AWSConfig) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&c.ConfigForOSDClusterCreation.AccountIDFilePath, "aws-account-id-file", c.ConfigForOSDClusterCreation.AccountIDFilePath, "File containing AWS account id")
+	fs.StringVar(&c.ConfigForOSDClusterCreation.AccessKeyFilePath, "aws-access-key-file", c.ConfigForOSDClusterCreation.AccessKeyFilePath, "File containing AWS access key")
+	fs.StringVar(&c.ConfigForOSDClusterCreation.SecretAccessKeyFilePath, "aws-secret-access-key-file", c.ConfigForOSDClusterCreation.SecretAccessKeyFilePath, "File containing AWS secret access key")
+	fs.StringVar(&c.Route53.AccessKeyFilePath, "aws-route53-access-key-file", c.Route53.AccessKeyFilePath, "File containing AWS access key for route53")
```

Comment:
I see there are 4 different flags around secrets manager. I also see the access key and secret access key are in two different new files.
Has it been considered making the different settings of the secret manager into its own config secret file?

Reply:
> I also see the access key and secret access key are in two different new files.

Yes - AppInterface basically gives you a secret par file when you ask for aws secret manager.